spec/run.sh

```diff
@@ -1,4 +1,54 @@
-for file in $(find spec -path '*/scripts/*.sh')
-do
-    bash "$file"
-done
+#!/bin/bash
+set -eu
+
+extract_json() {
+    local json_file="$1"
+    local key="$2"
+
+    local python_command="import json,sys;obj=json.load(sys.stdin);print(obj['${key}'])"
+    python3 -c "${python_command}" < "${json_file}"
+}
+
+extract_version() {
+    local text="$1"
+
+    [[ "${text}" =~ [0-9\.]+ ]]
+    echo "${BASH_REMATCH}"
+}
+
+setup_solc() {
+    echo "Setting up solc..." >&2
+    local solc_version="$(extract_version "$(solc --version)")"
+    local compiler_version="$(extract_version "$(extract_json .waffle.json compilerVersion)")"
+
+    echo "Found solc version ${solc_version} and Waffle compiler version ${compiler_version}." >&2
+    if [[ "${solc_version}" != "${compiler_version}" ]]; then
+        echo "Updating solc version to ${compiler_version}..." >&2
+        pip3 install solc-select
```

Comment:
You might want to consider installing it in python [virtual environment](https://docs.python.org/3/tutorial/venv.html). Same for certora-cli.

Reply:
Thanks! I'll look into this in a follow-up PR. Task created.